### PR TITLE
fix: port protobuf reader/writer to ts

### PR DESCRIPTION
**Pull request by achingbrain**
Ports just the parts of protobufjs we use to typescript to avoid the extra deps and integrates the changes from https://github.com/protobufjs/protobuf.js/pull/1557 to have native BigInt support.

### CODE REVIEW

**Comment by achingbrain** [comment]
This cannot be merged until https://github.com/nodejs/node/issues/44186 as resolved as it introduces a performance regression due to the use of small, short-lived ESM classes.

**Comment by wemeetagain** [comment]
@achingbrain can this be revisited?
It seems that the problem was more so related to the style of import (eg: named vs default vs namespace) rather than esm vs cjs.

**Comment by wemeetagain** [comment]
This will open the door for more performance improvements (eg: using a more optimized varint reader / writer)

This will also make adding additional types more straightforward (eg: 64 bit number types associated with https://github.com/ipfs/protons/issues/112)

**Comment by wemeetagain** [comment]
This appears to have no performance penalty vs the current master.
current master
```
pbjs x 8,051 ops/sec ±0.60% (91 runs sampled)
protons x 7,993 ops/sec ±0.45% (91 runs sampled)
protobuf.js x 8,182 ops/sec ±0.71% (91 runs sampled)
```

this PR
```
pbjs x 8,081 ops/sec ±0.69% (89 runs sampled)
protons x 8,353 ops/sec ±0.28% (95 runs sampled)
protobufjs x 8,376 ops/sec ±0.29% (95 runs sampled)
```

File: packages/protons-runtime/package.json [hunk 1]
```diff
@@ -148,8 +148,8 @@
     "release": "aegir release"
   },
   "dependencies": {
-    "protobufjs": "^7.0.0",
-    "uint8arraylist": "^2.3.2"
+    "uint8arraylist": "^2.3.2",
+    "uint8arrays": "^3.1.0"
```
Comment (wemeetagain) [inline packages/protons-runtime/package.json:152]: These can be updated

File: packages/protons-runtime/src/utils/longbits.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  public hi: number
+
+  constructor (lo: number, hi: number) {
+    // note that the casts below are theoretically unnecessary as of today, but older statically
+    // generated converter code might still call the ctor with signed 32bits. kept for compat.
+
+    /**
+     * Low bits.
+     *
+     * @type {number}
```
Comment (wemeetagain) [inline packages/protons-runtime/src/utils/longbits.ts:22]: I see a few of these typedoc type declarations that are superfluous

**Review by wemeetagain — APPROVED** [review]
lgtm

**Pull request merged**